google/yaml_validator.rb

```diff
@@ -42,17 +42,40 @@ def set_variable(value, property)
       instance_variable_set("@#{property}", value)
     end
 
-    private
+    # Does all validation checking for a particular variable.
+    # options:
+    # :default   - the default value for this variable if its nil
```

Comment:
I really like that this is specified in the same place as the check now and isn't just broken out above it.

Reply:
We had SO many validation functions. Which was part of the reason that we didn't use all of them to their fullest extent.